src/logic/basic.lean

```diff
@@ -155,9 +155,18 @@ assume ⟨h⟩, h.elim
 lemma congr_arg_heq {α} {β : α → Sort*} (f : ∀ a, β a) : ∀ {a₁ a₂ : α}, a₁ = a₂ → f a₁ == f a₂
 | a _ rfl := heq.rfl
 
-lemma plift.down_inj {α : Sort*} : ∀ (a b : plift α), a.down = b.down → a = b
+lemma ulift.down_injective {α : Sort*} : function.injective (@ulift.down α)
 | ⟨a⟩ ⟨b⟩ rfl := rfl
 
+@[simp] lemma ulift.down_inj {α : Sort*} (a b : ulift α) : a.down = b.down ↔ a = b :=
+⟨λ h, ulift.down_injective h, λ h, by rw h⟩
+
+lemma plift.down_injective {α : Sort*} : function.injective (@plift.down α)
+| ⟨a⟩ ⟨b⟩ rfl := rfl
+
+@[simp] lemma plift.down_inj {α : Sort*} (a b : plift α) : a.down = b.down ↔ a = b :=
+⟨λ h, plift.down_injective h, λ h, by rw h⟩
```

Comment:
```suggestion
plift.down_injective.eq_iff
```

Reply:
And same for the other one

Reply:
This doesn't actually work. `logic.basic` goes before in the import chain than `logic.function.basic`.

Reply:
Ah, I guess I forgot that `injective` is defined very very early, before most of its lemmas